server/controllers/match/bedSearchController.ts

```diff
@@ -1,41 +1,38 @@
 import type { Request, RequestHandler, Response } from 'express'
 import { BedSearchParametersUi } from '../../@types/ui'
+import paths from '../../paths/match'
 import { PersonService } from '../../services'
 import BedService from '../../services/bedService'
+import { DateFormats } from '../../utils/dateUtils'
+
+import { startDateFromParams } from '../../utils/matchUtils'
 
 export default class BedSearchController {
   constructor(private readonly bedService: BedService, private readonly personService: PersonService) {}
 
   search(): RequestHandler {
     return async (req: Request, res: Response) => {
       const query = Object.keys(req.query).length ? req.query : undefined
+      const body = Object.keys(req.body).length ? req.body : undefined
 
-      if (query) {
-        const {
-          durationDays,
-          maxDistanceMiles,
-          postcodeDistrict,
-          requiredPremisesCharacteristics,
-          requiredRoomCharacteristics,
-          startDate,
-          crn,
-        } = query
-
-        const bedSearchResults = await this.bedService.search(req.user.token, {
-          durationDays,
-          maxDistanceMiles,
-          postcodeDistrict,
-          requiredPremisesCharacteristics,
-          requiredRoomCharacteristics,
-          startDate,
-        } as BedSearchParametersUi)
+      if (query || body) {
+        const params = {
+          ...query,
+          ...body,
+        }
 
-        const person = await this.personService.findByCrn(req.user.token, crn as string)
+        params.startDate = startDateFromParams(params)
 
         res.render('match/search', {
           pageHeading: 'Find a bed',
-          bedSearchResults,
-          person,
+          bedSearchResults: await this.bedService.search(req.user.token, {
+            ...params,
+          } as BedSearchParametersUi),
+          person: await this.personService.findByCrn(req.user.token, params.crn as string),
+          formPath: paths.beds.search({}),
+          ...params,
+          expectedArrivalUi: params.expectedArrival && DateFormats.isoDateToUIDate(params.expectedArrival),
```

Comment:
As this is UI-specific, could we call it in the view itself? Would save the slightly awkward variable naming too.

Reply:
It also doesn't appear to be called in the template?

Reply:
[It is called](https://github.com/ministryofjustice/hmpps-approved-premises-ui/blob/eb0bc711566f5b82ed4e9b4f1e79da6d3b5981d1/server/views/match/search.njk#L40) but agree that it would be better to do the manipulation in the view